assembly/src/parsers/field_ops.rs

```diff
@@ -463,11 +495,9 @@ pub(super) fn parse_gte(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(),
 /// After these operations, the stack state will be: [2^a, ...].
 ///
 /// VM cycles per mode:
-/// - checked: 44 cycles
-/// - unchecked: 38 cycles
+/// - checked: 24 cycles
+/// - unchecked: 18 cycles
 pub(super) fn append_pow2_op(span_ops: &mut Vec<Operation>, checked_mode: bool) {
-    const MOST_SIGNIFICANT_BIT: u32 = 5;
-
     if checked_mode {
         // Checks if the top element of the stack is less than 64 or not. U32assert2 will
         // ensure if the element to which we are raising 2 to is u32 or not before u32div.
```

Comment:
Do we still need this check? It seems to me that if the value is not 64, this will be caught by the `assert` on line 530.